torch/csrc/distributed/spmd/event_impl.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <c10/util/Exception.h>
+#include <torch/csrc/distributed/spmd/event.h>
+
+// This file contains implementations of all specific Event classes.
+
+namespace torch {
+namespace distributed {
+namespace spmd {
+
+class PrepareModuleEvent : public Event {
+ public:
+  explicit PrepareModuleEvent(std::vector<at::Tensor> params)
```

Comment:
nit: pass by reference

Reply:
`params` takes an rvalue at call site and the same `params` is later moved to `params_`. IIUC, it won't introduce an additional copy, but explicitly marking it with `&&` is certainly more clear. Let me revise.